ddl/metadatalocktest/mdl_test.go

```diff
@@ -993,3 +993,115 @@ func TestMDLSavePoint(t *testing.T) {
 	tk.MustExec("commit")
 	tk.MustQuery("select * from t;").Check(testkit.Rows("1 <nil>"))
 }
+
+func TestMDLTableCreate(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+
+	tk := testkit.NewTestKit(t, store)
+	tkDDL := testkit.NewTestKit(t, store)
+
+	tk.MustExec("use test")
+	tk.MustExec("set global tidb_enable_metadata_lock=1")
+	tk.MustExec("create table t(a int);")
+	tk.MustExec("insert into t values(1);")
+
+	tk.MustExec("begin")
+	tk.MustQuery("select * from t;")
+	tk.MustGetErrCode("select * from t1;", mysql.ErrNoSuchTable)
+
+	tkDDL.MustExec("create table test.t1(a int);")
+
+	tk.MustGetErrCode("select * from t1;", mysql.ErrNoSuchTable)
```

Comment:
In RC， the second time， should t1 be seen?

Reply:
No for now, which can be optimized later.